azurerm/internal/services/containers/kubernetes_cluster_node_pool_resource_test.go

```diff
@@ -1349,6 +1371,26 @@ resource "azurerm_kubernetes_cluster_node_pool" "test" {
 `, r.templateConfig(data))
 }
 
+func (r KubernetesClusterNodePoolResource) upgradeSettingsConfig(data acceptance.TestData) string {
```

Comment:
since this test will be updating too, we can pass the other param in here:

```suggestion
func (r KubernetesClusterNodePoolResource) upgradeSettingsConfig(data acceptance.TestData, maxSurge string) string {
```